### PR TITLE
AP_GPS: UBlox: configure and log TIM-TM2

### PR DESCRIPTION
**Pull request by IamPete1**
This adds logging of TIM TM2 message useful for precise sync of signals to GPS time, via ext int pin on GPS. 

See: https://github.com/ArduPilot/ardupilot/issues/15428, https://github.com/ArduPilot/ardupilot/issues/10369, https://github.com/ArduPilot/ardupilot/issues/7892

Tested on m8p, I have verified the logged message is the same as that reported in u-center. 

image

I have not checked if this message is supported by all UBlox variants, or what happens if it is not supported, I presume the GPS will not respond to the request for message rate in that case. So this new message will not be added to `_unconfigured_messages` and will not cause a arming failure.

### CODE REVIEW

File: libraries/AP_GPS/AP_GPS_UBLOX.cpp [hunk 1]
```diff
@@ -723,6 +735,31 @@ void AP_GPS_UBLOX::log_mon_hw2(void)
 #endif
 }
 
+void AP_GPS_UBLOX::log_tim_tm2(void)
+{
+#if HAL_LOGGING_ENABLED
+    if (!should_log()) {
+        return;
+    }
+    struct log_UbxT pkt = {
```
Comment (peterbarker) [inline libraries/AP_GPS/AP_GPS_UBLOX.cpp:744]:
```suggestion
    const struct log_UbxT pkt {
```

**Review by tridge — CHANGES_REQUESTED** [review]
move to optional config, and only on >1M boards? also would be good to move RXM logging to be optional
better to use inline logging method so we don't consume a log slot for a rare msg

**Comment by IamPete1** [comment]
image

I left this logging for 9 and a half hours. A ~0.3Hz trigger signal connected to the ExtInt pin of the GPS and the CAM feedback pin. 

Firstly its clear that the `CAM.GPSTime` is useless. It is just the last received GPS time, there is no projection based on time since that message was received.  Unsurprisingly when the GPS rate is 5hz this results in a 0.2 second error range. I'm not sure why we also see a 17 second offset, could be some difference between timebase's used?

The final graph shows the CAM log time vs TM2 GPS time (offset to start at 0). This shows good agreement, although a linear drift. Works out to 2.533 ms per minute.

File: libraries/AP_GPS/AP_GPS_UBLOX.cpp [hunk 2]
```diff
@@ -723,6 +741,50 @@ void AP_GPS_UBLOX::log_mon_hw2(void)
...
+// @Field: MsR: Tow of rising edge
+// @Field: SubMsR: Millisecond fraction of tow of rising edge in nanoseconds
+// @Field: wnF: Week number of last falling edge
+// @Field: MsF: Tow of falling edge
+// @Field: SubMsF: Millisecond fraction of tow of falling edge in nanoseconds
+// @Field: accEst: Accuracy estimate
+
+    AP::logger().WriteStreaming("UBXT",
+        "TimeUS,I,ch,flags,count,wnR,MsR,SubMsR,wnF,MsF,SubMsF,accEst",
+        "s#s#----ss-sss",
```
Comment (tridge) [inline libraries/AP_GPS/AP_GPS_UBLOX.cpp:769]: wrong length?

File: libraries/AP_GPS/AP_GPS_UBLOX.cpp [hunk 3]
```diff
@@ -723,6 +741,50 @@ void AP_GPS_UBLOX::log_mon_hw2(void)
...
+        "F-----CI-CII",
```
Comment (tridge) [inline libraries/AP_GPS/AP_GPS_UBLOX.cpp:770]: is accEst mult really 1e-9?

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_GPS/AP_GPS_UBLOX.cpp (hunk 2 above)
Comment (IamPete1) [reply libraries/AP_GPS/AP_GPS_UBLOX.cpp]: yep, fixed

**Review by tridge — APPROVED** [review]

**Pull request merged**